### PR TITLE
Removes a lot of unused code

### PR DESCRIPTION
**Pull request by sobolevn**
After working on https://github.com/python/mypy/pull/11696 I was interested in knowing how many more unused code we have.

I am not suggesting that we should remove everything I found, all items must be carefully reviewed. Because they might be used by other people. But, I am sure we can totally remove **some** of the code I found.

### CODE REVIEW

**Comment by sobolevn** [comment]
TODO: `is_subtype_ignoring_tvars`

File: mypy/typeops.py [hunk 1]
```diff
@@ -44,25 +44,6 @@ def tuple_fallback(typ: TupleType) -> Instance:
...
-        return tuple_fallback(typ)
-    elif isinstance(typ, TypedDictType):
-        return typ.fallback
-    elif isinstance(typ, FunctionLike):
-        return typ.fallback
-    elif isinstance(typ, LiteralType):
-        return typ.fallback
-    else:
-        return None
-
```
Comment (97littleleaf11) [inline mypy/typeops.py:64]: I think we should keep this.

Comment (sobolevn) [reply mypy/typeops.py:64]: Was it used in some important place? Or is it popular in plugin code? 🤔

Comment (97littleleaf11) [reply mypy/typeops.py:64]: Personally speaking, this func helped me understand the effects of `fallback` and related code structure in mypy when I was a new contributor

Comment (sobolevn) [reply mypy/typeops.py:64]: @97littleleaf11 we can try to solve this via docs. Probably I can copy paste this function into some docstring. Thanks a lot for the idea! 👍

File: mypy/types.py [hunk 2]
```diff
@@ -903,6 +903,31 @@ class Instance(ProperType):
     """An instance type of form C[T1, ..., Tn].
 
     The list of type variables may be empty.
+
+    Several types has fallbacks to `Instance`. Why?
```
Comment (sobolevn) [inline mypy/types.py:907]: CC @97littleleaf11

**Comment by hauntsaninja** [comment]
This is great. Looks like some of the useful looking functions in there dated back to the old semantic analyser.

**Pull request merged**